packages/core/admin/ee/admin/src/pages/SettingsPage/pages/Users/components/ModalForm.tsx

```diff
@@ -13,18 +14,22 @@ export const ROLE_LAYOUT = [
     ? [
         [
           {
-            intlLabel: {
+            label: {
               id: 'Settings.permissions.users.form.sso',
               defaultMessage: 'Connect with SSO',
             },
             name: 'useSSORegistration',
-            type: 'bool',
-            size: {
-              col: 6,
-              xs: 12,
-            },
-          } as const,
+            type: 'boolean' as const,
+            size: 6,
+          },
         ],
       ]
     : []),
-] satisfies FormLayout[][];
+] satisfies Array<
+  Array<
+    Omit<InputProps, 'label'> & {
+      label: MessageDescriptor;
+      size: number;
+    }
+  >
+>;
```

Comment:
that type is a bit complex and you have almost the same one elsewhere. Do you think it could be extracted?

Reply:
it can, and it has been! ⭐